src/realm/util/string_buffer.hpp

```diff
@@ -134,14 +132,16 @@ inline const char* StringBuffer::data() const noexcept
 
 inline char* StringBuffer::c_str() noexcept
 {
+    static char zero = 0;
```

Comment:
@finnschiermer no, there will be a mutex'ed test of initialization upon each call to c_str() unless it's const... 

There was both a const and a non-const version of c_str(). What I have done now is to remove the non-const version